components/component_storage/CMakeLists.txt

```diff
@@ -36,6 +36,10 @@ add_hpx_component(component_storage
   DEPENDENCIES unordered_component
   ${_exclude_from_all_flag})
 
+if (TARGET hpx::apex)
+  target_link_libraries(component_storage_component PRIVATE hpx::apex)
+endif()
```

Comment:
Do we really need this now? Conceptually there shouldn't be any reason for a library that depends on HPX to see (and link) to APEX.